tests/test_pipelines_table_question_answering.py

```diff
@@ -0,0 +1,170 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from transformers.pipelines import Pipeline, pipeline
+from transformers.testing_utils import require_pandas, require_torch, require_torch_scatter
+
+from .test_pipelines_common import CustomInputPipelineCommonMixin
+
+
+@require_torch_scatter
+@require_torch
+@require_pandas
+class TQAPipelineTests(CustomInputPipelineCommonMixin, unittest.TestCase):
+    pipeline_task = "table-question-answering"
+    pipeline_running_kwargs = {
+        "padding": "max_length",
+    }
+    small_models = [
+        "lysandre/tiny-tapas-random-wtq",
+        "lysandre/tiny-tapas-random-sqa",
+    ]
+    large_models = ["nielsr/tapas-base-finetuned-wtq"]  # Models tested with the @slow decorator
+    valid_inputs = [
+        {
+            "table": {
+                "actors": ["brad pitt", "leonardo di caprio", "george clooney"],
+                "age": ["56", "45", "59"],
+                "number of movies": ["87", "53", "69"],
+                "date of birth": ["7 february 1967", "10 june 1996", "28 november 1967"],
+            },
+            "query": "how many movies has george clooney played in?",
+        },
+        {
+            "table": {
+                "actors": ["brad pitt", "leonardo di caprio", "george clooney"],
+                "age": ["56", "45", "59"],
+                "number of movies": ["87", "53", "69"],
+                "date of birth": ["7 february 1967", "10 june 1996", "28 november 1967"],
+            },
+            "query": ["how many movies has george clooney played in?", "how old is he?", "what's his date of birth?"],
+        },
+        {
+            "table": {
+                "Repository": ["Transformers", "Datasets", "Tokenizers"],
+                "Stars": ["36542", "4512", "3934"],
+                "Contributors": ["651", "77", "34"],
+                "Programming language": ["Python", "Python", "Rust, Python and NodeJS"],
+            },
+            "query": [
+                "What repository has the largest number of stars?",
+                "Given that the numbers of stars defines if a repository is active, what repository is the most active?",
+                "What is the number of repositories?",
+                "What is the average number of stars?",
+                "What is the total amount of stars?",
+            ],
+        },
+    ]
+
+    def _test_pipeline(self, nlp: Pipeline):
```

Comment:
Can we stop naming all pipelines `nlp`? It's super vague when here we know we have table qs pipeline (so for instance `table_querier` or something of the like would be better).